packages/wallets/src/features/cashier/modules/Transfer/hooks/useModifyTransferAccounts.ts

```diff
@@ -0,0 +1,187 @@
+import { useMemo } from 'react';
+import { useActiveWalletAccount, useAuthorize, useCurrencyConfig } from '@deriv/api';
+import { displayMoney } from '@deriv/api/src/utils';
+import { THooks } from '../../../../../types';
+import { getAccountName, getLandingCompanyNameOfMT5Account, getMarketType } from '../helpers';
+
+type TAccountType = THooks.TransferAccount['account_type'];
+type TMt5MarketType = ReturnType<typeof getMarketType>;
+
+/** A custom hook that enhances the transfer accounts response by adding additional properties for convenient UI rendering. */
+const useModifyTransferAccounts = (accounts?: THooks.TransferAccount[]) => {
```

Comment:
what does it mean "modifyTransferAccount"? its really hard to guess whats available out there by the name
also, that name enables this hook to became a bag for pretty much any field / any modifications, 
also I can imagine someone will need another version of account, so what they will use for name? useModifyTransferAccount2 and then useModifyTransferAccount3?

Reply:
if no good name can be found for this hook, that typically means, that there are too many responsibilities within one component (lack of S from SOLID). If thats the case, then might be worth separating e.g. visual aspects to the components themselves to limit responsibilities. Or figure out what kind of real-world object this represents. 

cc @markwylde-deriv

Reply:
cc @markwylde-deriv - wondering if it would be wise to define domain models in code somewhere and stick to them. Interface/type or even actual classes with getters/functions populated from backend. 
Anything which does not fit into domain model, probably then should be a helper function/component

Reply:
pls refer this [discussion](https://github.com/binary-com/deriv-app/pull/11419#discussion_r1393434639)